flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationOnlyCommandLineParser.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util;
+
+import org.apache.flink.runtime.entrypoint.FlinkParseException;
+import org.apache.flink.runtime.entrypoint.parser.CommandLineParser;
+import org.apache.flink.runtime.entrypoint.parser.ParserResultFactory;
+
+import org.apache.commons.cli.Options;
+
+import javax.annotation.Nonnull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Command line parser that parses only configuration related options.
+ * Unrelated options will be ignored.
+ */
+public class ConfigurationOnlyCommandLineParser<T> extends CommandLineParser<T> {
+
+	ConfigurationOnlyCommandLineParser(@Nonnull ParserResultFactory<T> parserResultFactory) {
+		super(parserResultFactory);
+	}
+
+	public T parse(@Nonnull String[] args) throws FlinkParseException {
+		final Options knownOptions = parserResultFactory.getOptions();
+
+		final List<String> filteredArgs = new ArrayList<>();
+		final Iterator<String> iter = Arrays.asList(args).iterator();
+
+		while (iter.hasNext()) {
+			String token = iter.next();
+			if (knownOptions.hasOption(token)) {
```

Comment:
As we do not have tests for this, not sure if this will work for:
`-Dsome.opt=value`
We probably have to split it manually (like put a space after -D).
See also `DefaultParser#isJavaProperty`.
We should also add tests for this.

Reply:
Maybe, it is better to implement this filtering separately in `BashJavaUtils` with `ClusterConfigurationParserFactory#getOptions ` before calling the parser to avoid inheritance complications.
Also not sure whether this filtering is always reusable/stable for all options.
`TaskManagerRunner#loadConfiguration` could be then reused after factoring it out to some common place.